cassiopeia/data/CassiopeiaTree.py

```diff
@@ -769,3 +806,78 @@ def relabel_nodes(self, relabel_map: Dict[str, str]) -> None:
 
         # reset cache because we've changed names
         self.__cache = {}
+
+    def get_dissimilarity_map(self):
+        """Gets the dissimilarity map.
+        """
+
+        return self.__dissimilarity_map.copy()
+
+    def set_dissimilarity_map(self, dissimilarity_map: pd.DataFrame):
+        """Sets the dissimilarity map variable in this object.
+        
+        Args:
+            dissimilarity_map: Dissimilarity map relating all N x N distances
+                between leaves.
+        """
+
+        self.__dissimilarity_map = dissimilarity_map.copy()
```

Comment:
If the keys in the `dissimilarity_map` don't match the keys in the `character_matrix`, when will the error be spotted?

Reply:
This is a great question, & especially difficult to say what the best approach is because the CassiopeiaTree need not have both inside of it.

But, I'm also not sure they need to have the same samples. The only place where this kind of error could be introduced is in the `set_dissimilarity_map` function, so maybe we can just raise a Warning if a character matrix exists and the `dissimilarity_map` does not have the same samples.

Reply:
I see, maybe some simple warning to catch an obvious mistake might be helpful? I don't know the details of the solver so I'll leave this as a suggestion, no need to implement it.

Reply:
That's exactly what I did -- I throw a warning if a character matrix exists and the dissimilarity map does not have the same keys.